node/executor/src/lib.rs

```diff
@@ -932,97 +945,28 @@ mod tests {
 		});
 
 		let mut tt = new_test_ext(COMPACT_CODE, false);
-		let block1 = construct_block(
-			&mut tt,
-			1,
-			GENESIS_HASH.into(),
-			vec![
-				CheckedExtrinsic {
-				signed: None,
-				function: Call::Timestamp(timestamp::Call::set(42)),
-				},
-				CheckedExtrinsic {
-					signed: Some((charlie(), 0)),
-					function: Call::System(system::Call::remark(vec![0; (IDEAL_TRANSACTIONS_WEIGHT*2) as usize])),
-				}
-			]
-		);
-		let block2 = construct_block(
-			&mut tt,
-			2,
-			block1.1.clone(),
-			vec![
-				CheckedExtrinsic {
-				signed: None,
-				function: Call::Timestamp(timestamp::Call::set(52)),
-				},
-				CheckedExtrinsic {
-					signed: Some((charlie(), 1)),
-					function: Call::System(system::Call::remark(vec![0; (IDEAL_TRANSACTIONS_WEIGHT*2) as usize])),
-				}
-			]
-		);
-
-		// execute a big block.
-		executor().call::<_, NeverNativeValue, fn() -> _>(
-			&mut t,
-			"Core_execute_block",
-			&block1.0,
-			true,
-			None,
-		).0.unwrap();
-
-		// weight multiplier is increased for next block.
-		runtime_io::with_externalities(&mut t, || {
-			let fm = System::next_weight_multiplier();
-			println!("After a big block: {:?} -> {:?}", prev_multiplier, fm);
-			assert!(fm > prev_multiplier);
-			prev_multiplier = fm;
-		});
-
-		// execute a big block.
-		executor().call::<_, NeverNativeValue, fn() -> _>(
-			&mut t,
-			"Core_execute_block",
-			&block2.0,
-			true,
-			None,
-		).0.unwrap();
-
-		// weight multiplier is increased for next block.
-		runtime_io::with_externalities(&mut t, || {
-			let fm = System::next_weight_multiplier();
-			println!("After a big block: {:?} -> {:?}", prev_multiplier, fm);
-			assert!(fm > prev_multiplier);
-		});
-	}
-
-	#[test]
-	fn weight_multiplier_decreases_on_small_block() {
-		let mut t = new_test_ext(COMPACT_CODE, false);
-
-		let mut prev_multiplier = WeightMultiplier::default();
-
-		runtime_io::with_externalities(&mut t, || {
-			assert_eq!(System::next_weight_multiplier(), prev_multiplier);
+		// NOTE: This assumes that system::remark has the default.
+		let num_to_exhaust = multiplier_ideal() * 2 / SimpleDispatchInfo::default().weigh_data(());
+		println!("++ Generating {} transactions to fill {} weight units", num_to_exhaust, multiplier_ideal() * 2);
```

Comment:
These are not leftover. Since these tests are quite hacky it is good to have them there in case of failure.